### DISCUSSION TITLE
How to handle SQLite3 type mismatch?

### DISCUSSION THREAD

**Discussion by jsalort** [SeaQL/sea-orm#2101] [Q&A]
I have some SQLite3 databases created with another software. Some of them have a type mismatch where a column declared as Integer actually contains a Real. This is legal in SQLite3 which does not enforce strong typing.
But this results in an error with SeaORM,
```
mismatched types; Rust type `core::option::Option<i64>` (as SQL type `INTEGER`) is not compatible with SQL type `REAL`.
```

How would you handle this ?

**Comment by tyt2y3** [comment]
May be change the entity definition to f64 or decimal ?

**Reply by jsalort** [reply]
Thank you for the reply, and I thought that this fixed the issue. Indeed, the program has been working fine with f64 in the entity definition.
However, now I get a Query error:
```
Query Error: error occurred while decoding column "timestamp": mismatched types; Rust type `core::option::Option<f64>` (as SQL type `REAL`) is not compatible with SQL type `INTEGER`
```

The reason is that the column is declared as INTEGER. The database (created with sqlalchemy and Python) contains floats, except for some values that are integer.
My understanding is that sqlite has a dynamic typing where if one tries to write another type, it will try to coerce, but will write anyway.
So when the timestamp happens to be integer, it is saved as INTEGER, when it is not round, it is saved as REAL.
So the type is not the same for all rows...

So the SeaORM query works fine, as long as all rows had non-integer timestamps...

Is there a solution ?
(besides fixing the database file, obviously)

i.e. is it possible to allow compatibility between Rust type f64 and SQL type INTEGER, so that the same code will work for all the rows ?
